src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (isEmpty()) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail = new Node<>(tail, value, null);
+            tail.previous.next = tail;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, size);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            head = new Node<>(null, value, head);
+            head.next.previous = head;
+            size++;
+        } else {
+            Node<T> actual = getNode(index);
+            Node newNode = new Node<>(actual.previous, value, actual);
+            newNode.previous.next = newNode;
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, size);
+        T result = getNode(index).value;
+        remove(index);
+        add(value, index);
+        return result;
```

Comment:
```suggestion
      Node<T> node = getNode(index);
        T oldValue = node.value;
        node.value = value;
        return oldValue;
```
this way you don't need to call two methods add and remove, because you just need to override value

Reply:
Done. Thank you!